pkg/controller/volume/pvcprotection/pvc_protection_controller.go

```diff
@@ -254,6 +256,26 @@ func (c *Controller) pvcAddedUpdated(obj interface{}) {
 	if protectionutil.NeedToAddFinalizer(pvc, volumeutil.PVCProtectionFinalizer) || protectionutil.IsDeletionCandidate(pvc, volumeutil.PVCProtectionFinalizer) {
 		c.queue.Add(key)
 	}
+	if old != nil {
+		oldPvc, ok := old.(*v1.PersistentVolumeClaim)
+		if !ok {
+			utilruntime.HandleError(fmt.Errorf("PVC informer returned non-PVC object: %#v", old))
+			return
+		}
+		if oldPvc.UID == pvc.UID {
+			// handled above for pvc
+			return
+		}
+		key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(oldPvc)
+		if err != nil {
+			utilruntime.HandleError(fmt.Errorf("Couldn't get key for Persistent Volume Claim %#v: %v", oldPvc, err))
+			return
+		}
+		if protectionutil.IsDeletionCandidate(oldPvc, volumeutil.PVCProtectionFinalizer) {
```

Comment:
if the old and new PVC UID differ, that means the old PVC does not exist any more. why would we care if the old one (which no longer exists) is a deletion candidate?

Reply:
This check is for the case where notification masks the deletion of a PVC followed by creation of another PVC.
For the prior PVC, PVCProtectionFinalizer needs to be removed.

Reply:
> For the prior PVC, PVCProtectionFinalizer needs to be removed.

The prior PVC no longer exists, so it has no finalizers and is not possible to update